Content.Server/RuntimeFun/SpeakOnExceptionSystem.cs

```diff
@@ -0,0 +1,89 @@
+using Content.Server.Chat.Systems;
+using Content.Server.Speech;
+using Content.Shared.Chat;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+using Robust.Shared.Timing;
+using Serilog.Events;
+
+namespace Content.Server.RuntimeFun;
+
+public sealed class SpeakOnExceptionSystem : EntitySystem
+{
+    [Dependency] private readonly ILogManager _log = default!;
+    [Dependency] private readonly ChatSystem _chat = default!;
+    [Dependency] private readonly IGameTiming _timing = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly IPrototypeManager _proto = default!;
+
+    // Special log handler that just saves the latest error.
+    private SpeakOnExceptionLogHandler _logHandler = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        _logHandler = new SpeakOnExceptionLogHandler();
+        _log.RootSawmill.AddHandler(_logHandler);
+
+        SubscribeLocalEvent<SpeakOnExceptionComponent, MapInitEvent>(OnMapInit);
+        SubscribeLocalEvent<SpeakOnExceptionComponent, TransformSpeechEvent>(OnTransformSpeech, before: [ typeof(AccentSystem) ]);
+    }
+
+    private void OnMapInit(Entity<SpeakOnExceptionComponent> ent, ref MapInitEvent args)
+    {
+        ent.Comp.NextTimeCanSpeak = _timing.CurTime;
+    }
+
+    public override void Update(float frameTime)
+    {
+        var log = _logHandler.LastLog;
+        if (log == null)
+            return;
+
+        var query = EntityQueryEnumerator<SpeakOnExceptionComponent>();
+
+        while (query.MoveNext(out var uid, out var comp))
+        {
+            if (_timing.CurTime < comp.NextTimeCanSpeak)
+                continue;
+
+            _chat.TrySendInGameICMessage(uid, CensorMessage(comp), InGameICChatType.Speak, ChatTransmitRange.Normal, true);
+
+            comp.NextTimeCanSpeak += comp.SpeechCooldown;
+        }
+
+        _logHandler.LastLog = null;
+    }
+
+    private void OnTransformSpeech(Entity<SpeakOnExceptionComponent> ent, ref TransformSpeechEvent args)
+    {
+        if (_random.Prob(ent.Comp.ChanceSpeakNoAccent))
+            args.Cancel();
+    }
```

Comment:
This will apply to all speech, not just error messages, no?

Reply:
This is not mentioned in the PR description and I do not think this is a good feature to have.